src/main/fc/boot.c

```diff
@@ -476,16 +480,16 @@ void init(void)
 #ifdef USE_ADC
     drv_adc_config_t adc_params;
 
-    adc_params.enableVBat = feature(FEATURE_VBAT);
-    adc_params.enableRSSI = feature(FEATURE_RSSI_ADC);
-    adc_params.enableCurrentMeter = feature(FEATURE_CURRENT_METER);
-    adc_params.enableExternal1 = false;
+    adc_params.channelMask =
+            (feature(FEATURE_VBAT) << ADC_CHANNEL1_BIT)
+            | (feature(FEATURE_RSSI_ADC) << ADC_CHANNEL2_BIT)
+            | (feature(FEATURE_CURRENT_METER) << ADC_CHANNEL3_BIT);
 #ifdef OLIMEXINO
-    adc_params.enableExternal1 = true;
+    adc_params.channelMask |= (1 << ADC_CHANNEL4_BIT);
 #endif
 #ifdef NAZE
     // optional ADC5 input on rev.5 hardware
-    adc_params.enableExternal1 = (hardwareRevision >= NAZE32_REV5);
+    adc_params.channelMask |= ((hardwareRevision >= NAZE32_REV5) << ADC_CHANNEL4_BIT);
```

Comment:
IMO the generated code is identical. Using conditional operator as bit value is a bit strange (but correct). Not sure, it is only my personal preference ..